lib/handler/file.c

```diff
@@ -304,6 +304,7 @@ static struct st_h2o_sendfile_generator_t *create_generator(h2o_req_t *req, cons
     }
             TRY_VARIANT(H2O_COMPRESSIBLE_BROTLI, "br", ".br");
             TRY_VARIANT(H2O_COMPRESSIBLE_GZIP, "gzip", ".gz");
+            TRY_VARIANT(H2O_COMPRESSIBLE_ZSTD, "zstd", ".zstd");
```

Comment:
For this part, I think it is a good idea to arrange the files in order of compression ratio (compression efficiency), starting with the file format with the highest compression ratio.
In other words, zstd should be listed first.
Because the code opens the file for the first variant exists.
Your current code looks only enable zstd when neither br nor gz are present.

Reply:
> In other words, zstd should be listed first.

zstd is not always smaller than br.
thats depends on setting and contents.
since zstd focusing on speed too.

for precompression, simply you can just leave smaller file. if you require more control for choose compress algorithms,  mruby could cover that.